Common/Data/UniverseSelection/BenchmarkUniverse.cs

```diff
@@ -0,0 +1,107 @@
+    /// This universe keeps the benchmark security and its subscription separate
+    /// from the algorithm securities. Universe selection will fire only once,
+    /// when the benchmark security is added to the universe.
+    /// </summary>
+    public class BenchmarkUniverse : Universe, INotifyCollectionChanged
+    {
+        /// <summary>
+        /// Event fired when a symbol is added or removed from this universe
+        /// </summary>
+        public event NotifyCollectionChangedEventHandler CollectionChanged;
+
+        /// <summary>
+        /// Gets the settings used for subscriptions added for this universe (unused in this universe)
+        /// </summary>
+        public override UniverseSettings UniverseSettings => null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BenchmarkUniverse"/> class
+        /// </summary>
+        /// <param name="configuration">The configuration used to resolve the data for universe selection</param>
+        public BenchmarkUniverse(SubscriptionDataConfig configuration) : base(configuration)
+        {
+        }
+
+        /// <summary>
+        /// Returns the benchmark symbol in this universe
+        /// </summary>
+        /// <param name="utcTime">The current UTC time</param>
+        /// <param name="data">The symbols to remain in the universe</param>
+        public override IEnumerable<Symbol> SelectSymbols(DateTime utcTime, BaseDataCollection data)
+        {
+            return Securities.Select(x => x.Key);
+        }
+
+        /// <summary>
+        /// Returns an existing security for this universe
+        /// </summary>
+        public override Security GetSecurity(Symbol symbol, IAlgorithm algorithm)
+        {
+            return Securities.Select(x => x.Value.Security).FirstOrDefault(x => x.Symbol == symbol);
+        }
+
+        /// <summary>
+        /// Adds the specified symbol to this universe
+        /// </summary>
+        /// <param name="utcTime">The current UTC time</param>
+        /// <param name="security">The symbol to be added to this universe</param>
+        /// <returns>True if the symbol was added, false if it was already present</returns>
+        public bool AddBenchmarkSecurity(DateTime utcTime, Security security)
+        {
+            if (AddMember(utcTime, security))
+            {
+                OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Add, security.Symbol));
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Event invocator for the <see cref="CollectionChanged"/> event
+        /// </summary>
+        /// <param name="e">The notify collection changed event arguments</param>
+        protected virtual void OnCollectionChanged(NotifyCollectionChangedEventArgs e)
+        {
+            var handler = CollectionChanged;
+            handler?.Invoke(this, e);
```

Comment:
no change required, but interestingly, when you do `instance?.Method()` the compiler expands it to:
``` csharp
var local = instance;
if (instance != null)
{
    instance.Method();
}
```
The compiler does this to avoid potential race conditions with setting `instance` to null after the null check but before the method invocation, so the saving to a local is redundant (and probably optimized away by the compiler anyway :) ).

Now, in the old days (before the null navigation operation `?.`) this was absolutely necessary given the potential for race conditions modifying the underlying storage (`MulticastDelegate`) to be null to clear the list of event handlers, but is no longer needed.